### PR TITLE
CAMEL-9707: Add ListDataSet and FileDataSet

### PR DESCRIPTION
**Pull request by hqstevenson**

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/component/dataset/FileDataSet.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    private String delimiter = "\\z";
+
+    private List<Object> defaultBodies;
+
+    public FileDataSet(String sourceFileName) throws IOException {
+        this(new File(sourceFileName));
+    }
+
+    public FileDataSet(File sourceFile) throws IOException {
+        this(sourceFile, "\\z");
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/component/dataset/FileDataSet.java]: Is \\z some special token for new lines on any platform? I have not seen this before?

**Comment by davsclaus** [comment]
Thanks has been merged, can you close?

File: camel-core/src/main/java/org/apache/camel/component/dataset/FileDataSet.java (hunk 1 above)
Comment (hqstevenson) [inline camel-core/src/main/java/org/apache/camel/component/dataset/FileDataSet.java]: It's a boundary matcher for the "end of input".  It's listed in the javadocs for java.util.regex.Pattern ( https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html ), so I hope it's platform portable ;-)

**Comment by hqstevenson** [comment]
Docs updated - closing PR

**Pull request closed without merging**